src/utils/daos.c

```diff
@@ -0,0 +1,870 @@
+	return rc;
+}
+
+static int
+cont_op_hdlr(struct cmd_args_s *ap)
+{
+	daos_handle_t		pool;
+	daos_handle_t		coh;
+	daos_cont_info_t	cont_info;
+	int			rc;
+	int			rc2;
+	enum cont_op		op;
+	const int		RC_PRINT_HELP = 2;
+
+	assert(ap != NULL);
+	op = ap->c_op;
+
+	/* TODO: create functions per container op */
+
+	if (uuid_is_null(ap->cont_uuid)) {
+		fprintf(stderr, "valid cont uuid required\n");
+		fflush(stderr);
+		rc = RC_PRINT_HELP;
+		goto bad_opt_free_mdsrv;
+	}
+
+	/*
+	 * all cont operations require a pool handle, lets make a
+	 * pool connection
+	 */
+	rc = daos_pool_connect(ap->pool_uuid, ap->group, ap->mdsrv,
+			       DAOS_PC_RW, &pool,
+			       NULL /* info */, NULL /* ev */);
+	if (rc != 0) {
+		fprintf(stderr, "failed to connect to pool: %d\n", rc);
+		goto bad_pool_connect;
+	}
+
+	if (op == CONT_CREATE) {
+		/* TODO: add use case: create by pool UUID + path (UNS) */
+		rc = daos_cont_create(pool, ap->cont_uuid, NULL, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to create container: %d\n", rc);
+			goto bad_cont_create;
+		}
+		fprintf(stdout, "Successfully created container "DF_UUIDF"\n",
+			DP_UUID(ap->cont_uuid));
+	}
+
+	if (op != CONT_DESTROY) {
+		rc = daos_cont_open(pool, ap->cont_uuid, DAOS_COO_RW, &coh,
+				    &cont_info, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "cont open failed: %d\n", rc);
+			goto bad_cont_open;
+		}
+	}
+
+
+	if (op != CONT_DESTROY) {
+		rc = daos_cont_close(coh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to close container: %d\n", rc);
+			goto bad_cont_close;
+		}
+	}
+
+	if (op == CONT_DESTROY) {
+		rc = daos_cont_destroy(pool, ap->cont_uuid, 1, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to destroy container: %d\n",
+				rc);
+			goto bad_cont_destroy;
+		}
+		fprintf(stdout, "Successfully destroyed container "DF_UUIDF"\n",
+			DP_UUID(ap->cont_uuid));
+	}
+
+	/* TODO implement the following:
+	 * op == CONT_LIST_OBJS
+	 * op == CONT_QUERY
+	 * op == CONT_STAT
+	 * op == CONT_GET_PROP
+	 * op == CONT_SET_PROP
+	 * op == CONT_LIST_ATTRS
+	 * op == CONT_DEL_ATTR
+	 * op == CONT_GET_ATTR
+	 * op == CONT_SET_ATTR
+	 * op == CONT_CREATE_SNAP
+	 * op == CONT_LIST_SNAPS
+	 * op == CONT_DESTROY_SNAP
+	 * op == CONT_ROLLBACK
+	 */
+
+bad_cont_destroy:
+bad_cont_close:
+bad_cont_open:
+bad_cont_create:
+	/* Pool disconnect  in normal and error flows: preserve rc */
+	rc2 = daos_pool_disconnect(pool, NULL);
+	if (rc2 != 0) {
+		fprintf(stderr, "Pool disconnect failed : %d\n", rc2);
+	}
+	if (rc == 0)
+		rc = rc2;
+bad_pool_connect:
+bad_opt_free_mdsrv:
+	daos_rank_list_free(ap->mdsrv);
+	return rc;
+}
+
+static int
+help_hdlr(struct cmd_args_s *ap)
+{
+	FILE *stream;
+
+	assert(ap != NULL);
+
+	stream = (ap->ostream != NULL) ? ap->ostream : stdout;
+
+	fprintf(stream, "\
+usage: daos RESOURCE COMMAND [OPTIONS]\n\
+resources:\n\
+	  pool             pool\n\
+	  container (cont) container\n\
+	  help             print this message and exit\n");
+
+	fprintf(stream, "\n\
```

Comment:
(style)  Avoid line continuations in quoted strings

Reply:
following model found in dmg.c and dcont.c